### PR TITLE
Refactoring with a small fix

### PR DESCRIPTION
**Pull request by codeout**
Scope fields varies from 1 to 5 according to [RFC3954](https://www.ietf.org/rfc/rfc3954.txt), only 5 types so I'd think that we can migrate both Option and Scope definitions into a single YAML. It could also make the code DRY.

```
Scope 1 Field Type
         The relevant portion of the Exporter/NetFlow process to which
         the Options Template Record refers.
         Currently defined values are:
            1 System
            2 Interface
            3 Line Card
            4 Cache
            5 Template
```

### CODE REVIEW

**Comment by codeout** [comment]
BTW, it's likely that we have something to fix in netflow v9 part, I'm preparing another pull request.

* Support multiple flowsets in a netflow v9 PDU
* Wrong key of template cache
* Support sampler cache
* Other fixes

**Comment by tagomoris** [comment]
Totally this looks fine.

**Comment by repeatedly** [comment]
File separation is introduced by this patch > https://github.com/repeatedly/fluent-plugin-netflow/pull/5
So merging is better?

**Comment by codeout** [comment]
So, how do you think, @junpei-yoshino ?

**Comment by repeatedly** [comment]
If there is no against opinion tomorrow, I will merge this patch.

File: lib/fluent/plugin/parser_netflow.rb [hunk 1]
```diff
@@ -229,20 +221,17 @@ def handle_v9_flowset_options_template(flowset, record)
...
-              fields += entry
-            end
-            template.option_fields.each do |field|
-              entry = netflow_field_for(field.field_type, field.field_length, @fields)
-              if ! entry
-                throw :field
+
+            ['scope', 'option'].each do |category|
+              template["#{category}_fields"].each do |field|
+                entry = netflow_field_for(field.field_type, field.field_length, category)
```
Comment (repeatedly) [inline lib/fluent/plugin/parser_netflow.rb]: passing `category` to `netflow_field_for` is correct?
3rd argument of `netflow_field_for` assumes array type?
https://github.com/repeatedly/fluent-plugin-netflow/blob/644f9b3d1da20522aa5145e410761629f4a2282a/lib/fluent/plugin/parser_netflow.rb#L311

Comment (repeatedly) [inline lib/fluent/plugin/parser_netflow.rb]: Ah, sorry. This patch updates `netflow_field_for` itself.
Ignore this comment.

Comment (codeout) [inline lib/fluent/plugin/parser_netflow.rb]: No problem :smile:

File: lib/fluent/plugin/parser_netflow.rb [hunk 2]
```diff
@@ -229,20 +221,17 @@ def handle_v9_flowset_options_template(flowset, record)
...
-                throw :field
-              end
-              fields += entry
-            end
-            template.option_fields.each do |field|
-              entry = netflow_field_for(field.field_type, field.field_length, @fields)
-              if ! entry
-                throw :field
+
+            ['scope', 'option'].each do |category|
```
Comment (repeatedly) [inline lib/fluent/plugin/parser_netflow.rb]: Moved to `V9_CATEGORIES` like constant is better to reduce temporary allocation.

File: lib/fluent/plugin/parser_netflow.rb [hunk 3]
```diff
@@ -229,20 +221,17 @@ def handle_v9_flowset_options_template(flowset, record)
...
+              template["#{category}_fields"].each do |field|
+                entry = netflow_field_for(field.field_type, field.field_length, category)
+                if ! entry
```
Comment (repeatedly) [inline lib/fluent/plugin/parser_netflow.rb]: How about `throw :field unless entry` ?

**Comment by junpei-yoshino** [comment]
I think merging yaml file is better for DRY.
Separating file is not must. #5

**Comment by repeatedly** [comment]
Okay, merged.

**Pull request merged**